test/request/reducers/RequestDetailRequestReducers.spec.ts

```diff
@@ -0,0 +1,190 @@
+import { IMessage, IMessageEnvelope } from '../../../src/request/messages/IMessageEnvelope';
+import { IMiddlewareStartPayload } from '../../../src/request/messages/IMiddlewareStartPayload';
+import { IMiddlewareEndPayload } from '../../../src/request/messages/IMiddlewareEndPayload';
```

Comment:
Thinking out loud, regarding conventions around "models" of various types, it would be nice if all models coming from the server had a consistent postfix and models that are constructed for the views have another post fix. Would make it really easy to spot what levels of data are being passed around where.